### PR TITLE
Refactor/markdown component split

### PR DESCRIPTION
**Pull request by xuzijie1995**
# Summary

Refactor markdown component to improve code organization and maintainability. This PR focuses on modularizing the markdown component structure and fixing linting errors.

**Motivation:**
The current `markdown.tsx` file has grown quite large (500+ lines) and contains multiple complex components. This refactoring improves code organization by:
1. Splitting components into separate modules
2. Fixing existing linting errors (removing unnecessary semicolons and brackets)
3. Optimizing ECharts rendering performance in CodeBlock component

**Changes Made:**
1. Restructured file organization:
   - Moved `markdown.tsx` to `markdown/index.tsx`
   - Created `markdown-blocks/` directory for individual block components
   - Extracted utility functions to `markdown-utils.ts`
   - Separated `ErrorBoundary` into its own file
   
2. Created separate files for block components:
   - `code-block.tsx`: Code syntax highlighting and ECharts rendering
   - `pre-code.tsx`: Pre-formatted code blocks
   - `video-block.tsx`: Video embedding
   - `audio-block.tsx`: Audio player
   - `script-block.tsx`: Script handling
   - `paragraph.tsx`: Text paragraphs
   - `img.tsx`: Image rendering
   - `link.tsx`: Link handling

# Testing Required

**Critical Areas Needing Testing:**
1. Markdown Rendering Scenarios:
   - [ ] Basic markdown text rendering
   - [ ] Code blocks with different languages
   - [ ] ECharts rendering and interactions
   - [ ] Image display and handling
   - [ ] Link functionality
   - [ ] Video and audio playback
   - [ ] SVG rendering
   - [ ] Error boundary functionality

2. Import Path Verification:
   - [ ] Verify all components using `@/app/components/base/markdown` still work
   - [ ] Check for any broken imports in the codebase
   - [ ] Validate barrel file exports

3. Performance Testing:
   - [ ] ECharts rendering performance
   - [ ] Memory usage with multiple instances
   - [ ] Load time impact

# Risks and Mitigations
- Risk: Some import paths might be missed during refactoring
  - Mitigation: Need comprehensive testing across the application
- Risk: Performance regression in certain scenarios
  - Mitigation: Performance testing needed before merge

# Next Steps
1. Complete thorough testing of all components
2. Add automated tests for critical functionality
3. Document any found issues or edge cases
4. Consider further code splitting based on testing results

# Screenshots

| Before | After |
|--------|-------|
| N/A (Code structure changes) | N/A (Code structure changes) |

follow up #https://github.com/langgenius/dify/pull/20161
# Checklist

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [] I've updated the documentation accordingly.
- [] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Help Needed
- Need team members to help test different scenarios
- Looking for feedback on component organization
- Performance testing assistance welcome

### CODE REVIEW

**Comment by crazywoola** [comment]
It has conflicts with your previous PR.

**Comment by xuzijie1995** [comment]
Still any conflicts? I've pushed the merge commit.

File: web/app/components/base/markdown-blocks/code-block.tsx [hunk 1]
```diff
@@ -1,34 +1,19 @@
...
-import MarkdownForm from '@/app/components/base/markdown-blocks/form'
-import MarkdownMusic from '@/app/components/base/markdown-blocks/music'
-import ThinkBlock from '@/app/components/base/markdown-blocks/think-block'
 import { Theme } from '@/types/app'
 import useTheme from '@/hooks/use-theme'
-import cn from '@/utils/classnames'
-import SVGRenderer from './svg-gallery'
+import SVGRenderer from '../svg-gallery' // Assumes svg-gallery.tsx is in /base directory
+import MarkdownMusic from '@/app/components/base/markdown-blocks/music'
+import ErrorBoundary from '@/app/components/base/markdown'
```
[the remark this answers is not in the document]
Comment (xuzijie1995) [reply web/app/components/base/markdown-blocks/code-block.tsx]: Fixed

File: web/app/components/base/markdown-blocks/link.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * Extracted from the main markdown renderer for modularity.
+ * Handles special rendering for "abbr:" type links for interactive chat actions.
+ */
+import React from 'react'
+import { useChatContext } from '@/app/components/base/chat/chat/context'
+
+const Link = ({ node, children, ...props }: any) => {
+  if (node.properties?.href && node.properties.href?.toString().startsWith('abbr')) {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const { onSend } = useChatContext()
```
[the remark this answers is not in the document]
Comment (xuzijie1995) [reply web/app/components/base/markdown-blocks/link.tsx]: Fixed

**Review by crazywoola — APPROVED** [review]
